File: Coding Club Beginner Projects/Lesson 5/Is Palindrome/isPalindrome.cpp
#include <iostream>
#include <string>

//Rudimentary implementation: 
//Doesn't take into account spaces, special characters or anything like that, but for now we'll use it to sbow tbe students how to make an isPalindrome function
std::string reverseString(std::string stringToReverse)
{
    std::string reversed = "";

    for (int i = stringToReverse.length() - 1; i >= 0; i--)
    {
        reversed += stringToReverse[i];
    }

    return reversed;
}

bool isPalindrome(std::string str)
{
    std::string reversedStr = reverseString(str);
    return str == reversedStr;
}

int main()
{
    std::string stringToCheck = "";
    std::cout << "Check if a string is a Palindrome!" << '\n';
    std::cout << "Please enter in the string: " << '\n';
    std::getline(std::cin, stringToCheck);
    std::cout << isPalindrome(stringToCheck);
}

File: Coding Club Beginner Projects/Lesson 5/Is Palindrome/is_palindrome.py
#Rudimentary implementation: 
#Doesn't take into account spaces, special characters or anything like that, but for now we'll use it to sbow tbe students how to make an isPalindrome function
def reverse_string(string_to_reverse):
    reversed = ""
    i = len(string_to_reverse)
    while i >= 0:
        reversed += string_to_reverse[i]
        i -= 1
    return reversed

def is_palindrome(str):
    reversed_str = reverse_string(str)
    return str == reversed_str

print("Check if a string is a Palindrome!")
string_to_check = input("Please enter in the string: ")
print(is_palindrome(string_to_check))